packages/ses/src/error/assert.js

```diff
@@ -153,12 +153,37 @@ const getLogArgs = ({ template, args }) => {
  */
 const hiddenMessageLogArgs = new WeakMap();
 
+// So each error tag will be unique.
+let errorTagNum = 0;
+
+/**
+ * @type {WeakMap<Error, string>}
+ */
+const errorTags = new WeakMap();
+
+/**
+ * @param {Error} err
+ * @param {string=} optErrorName
+ * @returns {string}
+ */
+const tagError = (err, optErrorName = err.name) => {
+  let errorTag = errorTags.get(err);
+  if (errorTag !== undefined) {
+    return errorTag;
+  }
+  errorTagNum += 1;
+  errorTag = `${optErrorName}#${errorTagNum}`;
+  errorTags.set(err, errorTag);
+  return errorTag;
+};
+
 /**
  * @type {AssertMakeError}
  */
 const makeError = (
   optDetails = redactedDetails`Assert failed`,
   ErrorConstructor = Error,
+  optErrorName = undefined,
```

Comment:
As discussed, the third positional argument is usually the point where we start to consider an options bag. I'm not sure what further options might be added in the future, and it adds maybe 15 characters of calling overhead, but if there are going to be any, now is the time to switch. 

I'm not sure if this function is likely to be called very much by user code (as `assert.error()`), rather than by just framework code (like `marshal.js` when deserializing a foreign-sourced Error). If it's only/mostly framework code, then the extra calling overhead doesn't matter much.

Reply:
Done. It is now a typed options bag with an option named `errorName`.